core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -230,6 +213,27 @@ object TopicCommand extends Logging {
     }
   }
 
+  private[admin] def describeAndPrintConfig(topic: String, partitionReplicas: List[(Int, Seq[Int])], configs: Properties): String = {
+    if (configs.size() != 0) {
+      val numPartitions = partitionReplicas.size
+      val replicationFactor = partitionReplicas.head._2.size
+      val configDescription = ("Topic:%s\tPartitionCount:%d\tReplicationFactor:%d\tConfigs:%s"
+        .format(topic, numPartitions, replicationFactor, configs.map(kv => kv._1 + "=" + kv._2).mkString(",")))
+      println(configDescription)
+      configDescription
+    } else ""
+  }
+
+  private def describePartitions(zkUtils: ZkUtils, liveBrokers: Set[Int], reportUnderReplicatedPartitions: Boolean, topic: String, topicPartitionAssignment: Map[Int, Seq[Int]], partitionReplicas: List[(Int, Seq[Int])]): Unit = {
+    val topicAndPartitionToLeaderAndIsr = topicPartitionAssignment.map(x =>
+      (topic, x._1) ->(zkUtils.getLeaderForPartition(topic, x._1), zkUtils.getInSyncReplicasForPartition(topic, x._1))
+    )
+    if (reportUnderReplicatedPartitions)
+      new UnderReplicatedPartitionsDescription(topic, partitionReplicas, liveBrokers, topicAndPartitionToLeaderAndIsr).describeAndPrint()
+    if (reportUnderReplicatedPartitions)
+      new UnavailablePartitionsDescription(topic, partitionReplicas, liveBrokers, topicAndPartitionToLeaderAndIsr).describeAndPrint()
```

Comment:
Something is wrong here. 
Do you want to report under replicated partitions or un-available ones?